lib/artirix_data_models/cached_action_adaptor.rb

```diff
@@ -44,10 +44,10 @@ def enabled?
   private
 
   def get_cached_result
-    return nil unless enabled?
+    raise "Unexpected NilClass returned from cache - case 2" unless enabled?
 
     c = cache_read
-    return nil unless c.present?
+    raise "Unexpected NilClass returned from cache - case 3" unless c.present?
```

Comment:
this will *ALWAYS* fail. You are raising a generic exception whenever something is not cached. Which means:
- if it is not cached, it will fail
- if it is cached, after it expires, it will fail